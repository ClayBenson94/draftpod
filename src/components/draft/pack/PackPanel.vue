<script>

import MtgCard from '@/components/draft/core/MtgCard.vue';
import UiPanel from '@/components/core/UiPanel.vue';

export default {
  name: 'PackPanel',
  props: {
    pack: {
      type: Array,
      default: null
    }
  },
  computed: {
    drag_source: () => "DRAG_SOURCE_PACK",
  },
  components: {
    MtgCard, UiPanel
  },
}
</script>

<template>
  <UiPanel class="pack-panel">
    <div v-if="pack">
      <MtgCard v-for="card in pack" :key="card.key" 
            :card="card" :drag_source="drag_source">
      </MtgCard>
    </div>
    <div v-else>
      Waiting for pack....
    </div>
  </UiPanel>  

</template>

<style>

.pack-panel {
  background: transparent;
}

.pack-panel .card {
  padding: 0;
}

.draft-page .pack-panel .card-body {
  padding: 0;
  background-color: transparent;
}

.pack-panel .mtgcard img {
  width: 11.5%;
  min-height: 16.05%;
  margin-bottom: 3px;
}

@media only screen and (max-width: 1000px) {
.pack-panel .mtgcard img {
  width: 10%;
  min-height: 13.95%;
} 
}

</style>

