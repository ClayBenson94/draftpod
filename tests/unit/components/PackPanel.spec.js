

import { mount } from '@vue/test-utils'
import PackPanel from '../../../src/components/draft/pack/PackPanel.vue'

import cards from '../data/cards.json'

import providers from '../util/providers'

describe('PackPanel.vue', () => {

  test('displays cards', () => {
    const wrapper = mount(PackPanel, {
      propsData: { pack: cards },
      provide: providers
    });
    const cardSpans = wrapper.findAll('.card-select-container .mtgcard');
    expect(cardSpans).toHaveLength(cards.length);

    // do a snapshot test
    expect(wrapper.element).toMatchSnapshot();
  });

});

