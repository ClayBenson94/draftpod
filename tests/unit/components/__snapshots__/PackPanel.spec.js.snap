// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackPanel.vue displays cards 1`] = `
<div
  class="card pack-panel card-select-panel"
>
  <!---->
   
  <div
    class="card-body"
  >
    <div
      class="card-select-container"
    >
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/3/a/3a6e2816-d50b-41a5-b503-faa58dc7c94a.jpg?1547235176"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/4/8/48a7ac34-ec4a-4571-abb1-bff1fb67c78f.jpg?1547234819"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/b/c/bcf9b57a-a759-4488-965a-651070cd2156.jpg?1547235085"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/7/4/74e6f7be-4493-4081-ac67-d782ab2b3723.jpg?1547235106"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/0/a/0ac65975-6ff5-4863-b7b7-d7dbea213b50.jpg?1547234665"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/7/c/7c77a6b1-ef06-4da5-8e86-a5204216cb77.jpg?1547235257"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/2/6/26f7e55d-d4c9-4755-ab87-a592ba3fb64f.jpg?1547579529"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/7/8/78cc4e02-949f-4879-9f32-7c33490d0b45.jpg?1547236136"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/3/4/3433a2c2-d252-4cd8-97e8-389875b2cda0.jpg?1547235610"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
      <span
        class="mtgcard mtgcard-draggable"
        draggable="true"
      >
        <img
          src="https://img.scryfall.com/cards/normal/front/4/c/4ca942d7-a3a3-429f-a159-fc2363d9bca6.jpg?1547235390"
        />
         
        <!---->
         
        <div
          style="position: fixed; top: -1000px;"
        />
      </span>
    </div>
  </div>
</div>
`;
